src/NJsonSchema.CodeGeneration.TypeScript/DataConversionGenerator.cs

```diff
@@ -84,21 +86,76 @@ private static object CreateModel(DataConversionParameters parameters)
 
                 //StringToDateCode is used for date and date-time formats
                 UseJsDate = parameters.Settings.DateTimeType == TypeScriptDateTimeType.Date,
-                StringToDateCode = parameters.Settings.DateTimeType == TypeScriptDateTimeType.Date ? "new Date" :
-                        (parameters.Settings.DateTimeType == TypeScriptDateTimeType.MomentJS ||
-                        parameters.Settings.DateTimeType == TypeScriptDateTimeType.OffsetMomentJS) &&
-                        typeSchema.Format == JsonFormatStrings.TimeSpan ? "moment.duration" :
-                    parameters.Settings.DateTimeType == TypeScriptDateTimeType.OffsetMomentJS ? "moment.parseZone" : "moment",
-                DateTimeToStringCode =
-                        (parameters.Settings.DateTimeType == TypeScriptDateTimeType.MomentJS ||
-                        parameters.Settings.DateTimeType == TypeScriptDateTimeType.OffsetMomentJS) &&
-                        typeSchema.Format == JsonFormatStrings.TimeSpan ? "format('d.hh:mm:ss.SS', { trim: false })" :
-                    parameters.Settings.DateTimeType == TypeScriptDateTimeType.OffsetMomentJS ? "toISOString(true)" : "toISOString()",
+                StringToDateCode = GetStringToDateTime(parameters, typeSchema),
+                DateTimeToStringCode = GetDatetimeToString(parameters, typeSchema),
```

Comment:
Hey @RicoSuter,

Wondering whether you could look at how I broke out these two lines as separate methods.  I changed them to switch statements from ternary statements.  I'm not sure that I necessarily covered all cases correctly, can you comment?

Reply:
Much easier to read :-)